src/test/java/net/datafaker/providers/base/FinanceTest.java

```diff
@@ -101,7 +101,8 @@ void discoverCard() {
     @RepeatedTest(100)
     void unionpayCard() {
         String creditCard = finance.creditCard(CreditCardType.UNIONPAY).replace("-", "");
-        assertThat(creditCard).startsWith("62").hasSize(16);
+        assertThat(creditCard).hasSize(16);
+        assertThat(creditCard.startsWith("62") || creditCard.startsWith("64") || creditCard.startsWith("65") || creditCard.startsWith("81")).isTrue();
```

Comment:
Such checks have one problem: when this test fails, you cannot see the actual credit card number in the error message. You will only see "Test failed. Expected: true, actual: false".

(But the previous code also had the same problem)

Reply:
Fixed in #1643